src/lighteval/models/transformers/base_model.py

```diff
@@ -833,18 +837,60 @@ def greedy_until(
                     padded=[sum(mask == 0) for mask in tokenized["attention_mask"]],
                 )
 
-                cur_reponses = self._generate(
+                cur_reponses = self._generate_cached(
                     batch=prepared_batch,
                     max_new_tokens=max_new_tokens,
                     stop_tokens=stop_tokens,
                     returns_logits=returns_logits,
                     num_samples=num_samples,
                     do_sample=do_sample,
                 )
+
                 results.extend(cur_reponses)
 
         return dataset.get_original_order(results)
 
+    def _generate_cached(
+        self,
+        batch: Batch,
+        max_new_tokens: int,
+        stop_tokens: list[str],
+        returns_logits: Optional[bool] = False,
+        num_samples: Optional[int] = 1,
+        do_sample: Optional[bool] = False,
+    ) -> list[GenerativeResponse]:
+        # Create a cache key from the inputs
+        cache_key = (
+            hashlib.sha256(str(batch.input_ids.tolist()).encode()).hexdigest(),
+            max_new_tokens,
+            tuple(stop_tokens),
+            returns_logits,
+            num_samples,
+            do_sample,
+        )
+
+        # Try to get from cache first
+        cached_responses = self._generation_cache.get(cache_key)
+        if cached_responses is not None:
+            logger.info(f"Cache hit for batch {cache_key}")
```

Comment:
we wont want to log every hit for the cache. We can keep it as debug log though

Reply:
Sounds good to me.